File: m223-punchclock-master/src/main/java/ch/zli/m223/punchclock/domain/Place.java
package ch.zli.m223.punchclock.domain;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Place {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;
    @Column(name = "application_user_id")
    private Long applicationUserId;
    @Column(name = "e_icon")
    private EIcon eIcon;
    @Column(name="name")
    private String name;
}

File: m223-punchclock-client/src/app/model/Place.ts
import {EIcon} from "./EIcon";

export class Place {
  private _id:number;
  private _applicationUserId: number;
  private _eIcon: EIcon;
  private _name: string;


  constructor(id: number, applicationUserId: number, eIcon: EIcon, name: string) {
    this._id = id;
    this._applicationUserId = applicationUserId;
    this._eIcon = eIcon;
    this._name = name;
  }


  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get applicationUserId(): number {
    return this._applicationUserId;
  }

  set applicationUserId(value: number) {
    this._applicationUserId = value;
  }

  get eIcon(): EIcon {
    return this._eIcon;
  }

  set eIcon(value: EIcon) {
    this._eIcon = value;
  }

  get name(): string {
    return this._name;
  }

  set name(value: string) {
    this._name = value;
  }
}
